bindings/experimental/distrdf/python/DistRDF/Backends/Base.py

```diff
@@ -170,27 +170,34 @@ def mapper(current_range):
                 # We assume 'end' is exclusive
                 chain.SetCacheEntryRange(start, end)
 
-                # Gather information about friend trees
-                friend_info = current_range.friend_info
-                if friend_info:
-                    # Zip together the treenames of the friend trees and the
-                    # respective file names. Each friend treename can have
-                    # multiple corresponding friend file names.
-                    tree_files_names = zip(
-                        friend_info.friend_names,
-                        friend_info.friend_file_names
+                # Gather information about friend trees. Check that we got an
+                # RFriendInfo struct and that it's not empty
+                if (current_range.friend_info is not None and
+                    not current_range.friend_info.fFriendNames.empty()):
+                    # Zip together the information about friend trees. Each element if of the form:
+                    # (name, alias), (file1.root, file2.root, ...), (subname1, subname2, ...)
```

Comment:
So the `subnames` are the names of the friend tree in each file in the chain, but if instead of a chain it's a tree there are no subnames?
Would it be good to add a paragraph about the subnames here?

Reply:
Yes you are right. If the friend is a `TTree` it will not have sub names so the corresponding vector in `friend_info.fFriendChainSubNames` will be empty. You are also right that I should add some extra comments here :D